### ISSUE TITLE
Add option to disable / change language of spell check

### ISSUE THREAD

**Issue by tomaskavalek** [danielbuechele/goofy#442]
After last update – so I am using version 3.4.1.

Please, allow to disable spellcheck in the app. I am writing a lot of messages in czech language, so every word is with red line below, and every next word slowing down the message box. Allow to choose the language or disabling this spellcheck.

Thanks a lot.

**Comment by tomaskavalek** [comment]
Probably this [piece of code](https://github.com/danielbuechele/goofy/blob/38d10885d00825addebcfb312be966a4f9fd087a/app/src/fb.js#L325).

**Comment by jklp** [comment]
Added a new preferences pane to Goofy - where I've given the option for the user to disable the spell checker.

I've also added a drop down to select between the two different types of english or use the system locale (might work in some instances?).  Will expand to include more languages in future but hopefully the system one should work for now.

**Comment by jklp** [comment]
Ugh that's frustrating ... seems to work on my local machine though changing languages on the github binary doesn't seem to work ... re-opening so we can look at in next milestone

**Comment by tomaskavalek** [comment]
I have downloaded version 3.4.2 and everything works as I expected.

If I enable spell checker and keep select as System, so app use English spell-checker on my Czech locale. But it's due to missing support from spell-checker – English only.

**Comment by tomaskavalek** [comment]
If I uncheck spell checker, it's disabled. Good work I think.

**Comment by jklp** [comment]
Yeah sorry - was hoping would be a little cleaner but got too excited with the preferences pane I didn’t 100% let it bake before releasing. 

More than happy to accept PRs / feedback and now with preferences pane hopefully in good shape to make Goofy a little more configurable.

**Comment by jklp** [comment]
Now using `electron-spellchecker` to do spell checking which seems a lot more reliable.

**Comment by tomaskavalek** [comment]
Still not working. See [https://github.com/electron-userland/electron-spellchecker/issues/120](https://github.com/electron-userland/electron-spellchecker/issues/120). Have you tested it?

**Comment by jklp** [comment]
Yep, am located in Australia so I use UK English vs US and spelling errors have been highlighting correctly for me

Unfortunately I don't speak any other languages but I did do some quick testing with French and German via Google Translate which seemed to work - is it not working for you? 

By the way you'll need to restart Goofy for the language change to work (I haven't put in hot reloading of any settings yet)

**Comment by jklp** [comment]
Have confirmed, looks like problem upstream with `electron-spellchecker` - other issues below:

https://github.com/electron-userland/electron-spellchecker/issues/85
https://github.com/electron-userland/electron-spellchecker/issues/61

Have set spell checking to disabled by default as I don't think should remove entire feature if some users will find it helpful - have added doc to let users know the feature isn't 100% working though so hopefully clears things up.  

Haven't closed issue but moved out of 3.4.3 - lets revisit when we get some answers from `electron-spellchecker` (or find another lib which does the job)

**Comment by Palivec** [comment]
Hi ! I have a similar problem here !

My system language is English, but I write lot of Czech text too. Altough the spell check is disabled in Goofy preferences window, the app still **checks** the spelling, and only in English ! That means that all the Czech text is marked as incorrect, and it can't be disabled.

Moreover, changing Goofy spell option several times led to a blank app window and I had to delete all its ~/Library files. The option doesn't work for me at all.

I use Goofy 3.4.3 and MacOS 10.14.4

**Comment by jklp** [comment]
@Palivec Hi did you restart Goofy after disabling Spell check in preferences?  I just tried disabling on my end and typing in some Czech and no spelling errors are highlighted.

**Comment by jklp** [comment]
Looks like Electron supports spell checking natively now!

https://github.com/electron-userland/electron-spellchecker/issues/85#issuecomment-560006783

**Comment by jklp** [comment]
Just double checked, only supported in Electron 8 (still in beta) - will look into again once Electron 8 is out
